src/core/services-impl/src/main/java/it/geosolutions/geostore/services/ResourceServiceImpl.java

```diff
@@ -491,72 +493,82 @@ public long getCountByFilter(SearchFilter filter)
     }
 
     /**
-     * @param list
+     * @param resources
+     * @param user
      * @return List<ShortResource>
      */
-    private List<ShortResource> convertToShortList(List<Resource> list, User authUser) {
-        List<ShortResource> swList = new LinkedList<>();
-        for (Resource resource : list) {
-            ShortResource shortResource = new ShortResource(resource);
-
-            // ///////////////////////////////////////////////////////////////////////
-            // This fragment checks if the authenticated user can modify and delete
-            // the loaded resource (and associated attributes and stored data).
-            // This to inform the client in HTTP response result.
-            // ///////////////////////////////////////////////////////////////////////
-            boolean resourceCanBeListed = true;
-            if (authUser != null) {
-                if (authUser.getRole().equals(Role.ADMIN)) {
-                    shortResource.setCanEdit(true);
-                    shortResource.setCanDelete(true);
-                } else {
-                    boolean authUserIsOwner = false;
-                    for (SecurityRule rule : resource.getSecurity()) {
-                        User owner = rule.getUser();
-                        UserGroup userGroup = rule.getGroup();
-                        if (owner != null) {
-                            if (owner.getId().equals(authUser.getId())) {
-                                authUserIsOwner = true;
-                                if (rule.isCanWrite()) {
-                                    shortResource.setCanEdit(true);
-                                    shortResource.setCanDelete(true);
-
-                                    break;
-                                }
-                            }
-                        } else if (userGroup != null) {
-                            List<String> groups = extractGroupNames(authUser.getGroups());
-                            if (groups.contains(userGroup.getGroupName())) {
-                                if (rule.isCanWrite()) {
-                                    shortResource.setCanEdit(true);
-                                    shortResource.setCanDelete(true);
-
-                                    break;
-                                }
-                            }
-                        }
-                    }
-                    if (!authUserIsOwner) resourceCanBeListed = resource.isAdvertised();
+    private List<ShortResource> convertToShortList(List<Resource> resources, User user) {
+        return resources.stream()
+                .filter(r -> isResourceCanBeListed(user, r))
+                .map(r -> createShortResourceWithPermissions(user, r))
+                .collect(Collectors.toList());
+    }
```

Comment:
Overall, streaming is fine here. Just watch out for repeated database calls in .filter(...) or .map(...). If isResourceCanBeListed(...) or createShortResourceWithPermissions(...) each cause multiple DB queries, you might degrade performance.

Confirm that you’re not repeatedly calling resourceService.getSecurityRules(r.getId()) inside this stream for many resources. You might want to batch fetch or join if performance becomes an issue.

Reply:
Fixed. Thanks!